### ISSUE TITLE
Difference in setting nested model and collection attribute ( with fetch for exemple )

### ISSUE THREAD

**Issue by pgayvallet** [dhruvaray/backbone-associations#29]
I'm facing a problem when fetching model with nested attributes cause the nested model to loose reference (be overrided) : 

The use case is the following : 

Model A having a nested Model B as attribute "nested" ( A.get("nested") -> B )

A.fetch()
A.get("nested").on("any event", anycallback)
A.fetch()
-> nested have been overrided so the attached event is now attach to the old "zombie" model B.

This is easily explaided by the snippet in setAttr : 

``` javascript
else if (relation.type === Backbone.One && relatedModel) {
    data = val instanceof AssociatedModel ? val : new relatedModel(val);
        attributes[relationKey] = data;
    }
```

But the strangest thing is that the behaviour in setting a collection is not the same : you are using a reset is the value is not a collection:

``` javascript
if (!this.attributes[relationKey]) {
    data = collectionType ? new collectionType() : this._createCollection(relatedModel);
    data.add(val, relationOptions);
    attributes[relationKey] = data;
} else {
        this.attributes[relationKey].reset(val, relationOptions);
    delete attributes[relationKey];
}
```

I understand that the behaviour is tricky because you need to handle the cases where the nested models are manually sets ( by A.set("nested", B) for exemple) and where the dev is expecting the nested model to be set as it is. But at least the setAttr behaviour should be the same for a nested model and a nested collection in the case the attribute is not a backbone model or collection ( array or object )

The fix is easy : 

just replace

``` javascript
else if (relation.type === Backbone.One && relatedModel) {
    data = val instanceof AssociatedModel ? val : new relatedModel(val);
    attributes[relationKey] = data;
}
```

by 

``` javascript
else if (relation.type === Backbone.One && relatedModel) {
    if (val instanceof BackboneModel) {
        data = val;
        attributes[relationKey] = data;
    } else {
        if (!this.attributes[relationKey]) {
            data = val instanceof AssociatedModel ? val : new relatedModel(val);
            attributes[relationKey] = data;
        } else {
            this.attributes[relationKey].set(val, relationOptions);
            delete attributes[relationKey];
        }
    }
}
```

**Comment by dhruvaray** [comment]
I have fixed it with commit 54ce3dc4b8b6ff980607bb07bbccfd5645e781f3. I have decided to make collection attribute consistent with model attribute - rather than the other way around.

**Comment by dhruvaray** [comment]
@wayofspark  > The possibility of the server response updating the model made it compelling to re-visit the issue again. Thanks for the initial code snippet though. It wouldn't work in that form because of eventing issues. But you have the effect you were seeking with this commit now. You can look at issue #31 for a similar thread and discussion.
